### ISSUE TITLE
Module cannot be upgraded from 1.2.2 to 2.2.0 without first loading 2.0.x

### ISSUE THREAD

**Issue by tparkercbn** [leoarnold/puppet-cups#139]
**Given** Debian 8 with Puppet 4.12.1 and CUPS 1.7.5-11+deb8u4

**When** I apply the manifest

```puppet
include '::cups'

# Your manifest here
    # Create all printers listed in $printers::cups_queues then restart cups
    $queue_defaults = {
        notify  => Exec['RestartCupsAfter'],
    }

    create_resources('cups_queue', $printing::cups_queues, $queue_defaults)

    exec {'RestartCupsAfter':
        command         => '/bin/systemctl restart cups.service',
        refreshonly     => true,
    }

**In order to** Create a new printer class and 2 printers.

**Then** I get the error message:

Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Type-Name, Could not autoload puppet/type/cups_queue: /etc/puppetlabs/code/environments/production/modules/cups/lib/puppet/type/cups_queue.rb:104: unknown type of %string
      return "created a #{should_to_s(new_value)}" unless %i[class printer].include?(current_value)
                                                            ^ at /etc/puppetlabs/code/environments/production/modules/cups/manifests/queues/default.pp:24:18 on node 0060ef20812e-dmpgd.dmvlt.ls.cbn

If I upgrade from 1.2.2 to 2.0.3 then 2.2.2 using r10k and calling a cache flush on the puppet master it all works just fine.

**Comment by tparkercbn** [comment]
This may be a bug in puppet more than this module, but I have not run into it anywhere else and it's very strange.

**Comment by tparkercbn** [comment]
I see you have released a 2.2.1 in the forge.  Testing that one now.

**Comment by tparkercbn** [comment]
No issues with the 2.2.1 release.  Something about those  Array percent literals was messing things up.
